tstate/tstate_view.go

```diff
@@ -161,21 +161,8 @@ func (ts *TStateView) KeyOperations() (map[string]uint16, map[string]uint16) {
 }
 
 // checkScope returns whether [k] is in scope and has appropriate permissions.
-func (ts *TStateView) checkScope(_ context.Context, k []byte, perm state.PermissionBit) bool {
-	permission, exists := ts.scope[string(k)]
-	if !exists {
-		return false
-	}
-	switch perm {
-	case state.Read:
-		return permission.Read
-	case state.Allocate:
-		return permission.Allocate
-	case state.Write:
-		return permission.Write
-	default:
-		return false
-	}
+func (ts *TStateView) checkScope(_ context.Context, k []byte, perm state.Permissions) bool {
+	return ts.scope[string(k)].Has(perm)
```

Comment:
if the key is not returned, does this work?

Reply:
yes

Reply:
I suppose it just defaults to 0?

Reply:
Yeah - the lookup defaults to returning `None` permissions. `None.Has(anythingOtherThanNone)` returns `false`.